client/components/social-buttons/google.js

```diff
@@ -0,0 +1,84 @@
+/**
+ * External dependencies
+ */
+import React, { Component, PropTypes } from 'react';
+import SocialLogo from 'social-logos';
+import { loadScript } from 'lib/load-script';
+
+export default class GoogleLoginButton extends Component {
+	static propTypes = {
+		clientId: PropTypes.string.isRequired,
+		scope: PropTypes.string,
+		fetchBasicProfile: PropTypes.bool,
+		responseHandler: PropTypes.func.isRequired,
+	};
+
+	static defaultProps = {
+		scope: 'profile',
+		fetchBasicProfile: true,
+	};
+
+	constructor( props ) {
+		super( props );
+		this.googleAuth = null;
+		this.handleClick = this.handleClick.bind( this );
+	}
+
+	componentWillMount() {
+		this.loadGoogleAuth();
```

Comment:
By not storing the promise here, we'll initiate a second request in `handleClick`